modules/llama_cpp_plugin/src/compiled_model.cpp

```diff
@@ -0,0 +1,123 @@
+// Copyright (C) 2024 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+
+#include "compiled_model.hpp"
+
+#include <fstream>
+#include <memory>
+#include <openvino/op/constant.hpp>
+#include <openvino/opsets/opset13.hpp>
+#include <openvino/runtime/properties.hpp>
+#include <openvino/util/log.hpp>
+#include <thread>
+
+#include "infer_request.hpp"
+#include "plugin.hpp"
+
+namespace ov {
+namespace llama_cpp_plugin {
+
+LlamaCppModel::LlamaCppModel(const std::shared_ptr<ov::Model>& model,
+                             const std::shared_ptr<const ov::IPlugin>& plugin,
+                             const ov::SoPtr<ov::IRemoteContext>& context,
+                             const std::shared_ptr<ov::threading::ITaskExecutor>& task_executor)
+    : ICompiledModel(model, plugin, context, task_executor) {
+    OPENVINO_THROW_NOT_IMPLEMENTED("Currently only direct GGUF file loading is "
+                                   "supported for the LLAMA_CPP* plugins");
+}
+
+LlamaCppModel::LlamaCppModel(const std::shared_ptr<ov::Model>& ov_model,
+                             std::istream& input_stream,
+                             const std::shared_ptr<const IPlugin>& plugin)
+    : ICompiledModel(ov_model, plugin) {
+    OPENVINO_THROW_NOT_IMPLEMENTED("Currently only direct GGUF file loading is "
+                                   "supported for the LLAMA_CPP* plugins");
+}
+
+LlamaCppModel::~LlamaCppModel() {
+    llama_free(m_llama_ctx);
+    llama_free_model(m_llama_model_ptr);
+    llama_backend_free();
+}
+
+LlamaCppModel::LlamaCppModel(const std::string& gguf_fname, const std::shared_ptr<const IPlugin>& plugin)
+    : ICompiledModel(nullptr, plugin),
+      m_gguf_fname(gguf_fname) {
+    OPENVINO_DEBUG << "llama_cpp_plugin: loading llama model directly from GGUF... " << std::endl;
+    llama_model_params mparams = llama_model_default_params();
+    mparams.n_gpu_layers = 99;
+    m_llama_model_ptr = llama_load_model_from_file(gguf_fname.c_str(), mparams);
+    llama_context_params cparams = llama_context_default_params();
+    cparams.n_threads =
+        std::thread::hardware_concurrency();  // TODO (vshampor): reuse equivalent setting defined by OV API
```

Comment:
OpenVINO has https://github.com/openvinotoolkit/openvino/blob/master/src/inference/include/openvino/runtime/properties.hpp#L1156

But maybe it's not important for current state